adapters/nextmillennium/nextmillennium.go

```diff
@@ -0,0 +1,162 @@
+type adapter struct {
+	endpoint string
+}
+
+type NextMillenniumBidRequest struct {
+	ID   string `json:"id"`
+	Test uint8  `json:"test,omitempty"`
+	Ext  struct {
+		Prebid struct {
+			StoredRequest struct {
+				ID string `json:"id"`
+			} `json:"storedrequest"`
+		} `json:"prebid"`
+	} `json:"ext"`
+}
+
+//MakeRequests prepares request information for prebid-server core
+func (adapter *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	resImps, err := getImpressionsInfo(request.Imp)
+	if len(err) > 0 {
+		return nil, err
+	}
+
+	result := make([]*adapters.RequestData, 0, len(resImps))
+	for _, imps := range resImps {
+		bidRequest, err := adapter.buildAdapterRequest(request, imps, request.Imp)
+		if err != nil {
+			return nil, []error{err}
+		} else {
+			result = append(result, bidRequest)
+		}
+	}
+	return result, nil
+}
+
+// getImpressionsInfo checks each impression for validity and returns impressions copy with corresponding exts
+func getImpressionsInfo(imps []openrtb2.Imp) (resImps []*openrtb_ext.ImpExtNextMillennium, errors []error) {
+	for _, imp := range imps {
+		impExt, err := getImpressionExt(&imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		if err := validateImpression(impExt); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		resImps = append(resImps, impExt)
+	}
+
+	return
+}
+
+func validateImpression(impExt *openrtb_ext.ImpExtNextMillennium) error {
+	if impExt.PlacementID == "" {
+		return &errortypes.BadInput{Message: "No valid placement provided"}
+	}
+	return nil
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtNextMillennium, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	var nextMillenniumExt openrtb_ext.ImpExtNextMillennium
+	if err := json.Unmarshal(bidderExt.Bidder, &nextMillenniumExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+
+	return &nextMillenniumExt, nil
+}
+
+func (adapter *adapter) buildAdapterRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ImpExtNextMillennium, imps []openrtb2.Imp) (*adapters.RequestData, error) {
+	newBidRequest := createBidRequest(prebidBidRequest, params, imps)
+
+	reqJSON, err := json.Marshal(newBidRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     adapter.endpoint,
+		Body:    reqJSON,
+		Headers: headers}, nil
+}
+
+func createBidRequest(prebidBidRequest *openrtb2.BidRequest, params *openrtb_ext.ImpExtNextMillennium, imps []openrtb2.Imp) *NextMillenniumBidRequest {
+	bidRequest := NextMillenniumBidRequest{
```

Comment:
Similar to my previous comment above. I am not sure how your bidder service would be able to process a request just with the bid request ID and the placement ID? The `imps` variable that you're passing into this function is unused.